tensorboard/plugins/profile/profile_plugin.py

```diff
@@ -427,18 +427,16 @@ def data_route(self, request):
   @wrappers.Request.application
   def capture_route(self, request):
     service_addr = request.args.get('service_addr')
-    try:
-      duration = int(request.args.get('duration'))
-    except TypeError:
-      # Returns code=200 to show error message in UI.
-      return http_util.Respond(request, {'error': 'invalid duration.'},
-                               'application/json', code=200)
+    duration = int(request.args.get('duration'))
```

Comment:
Is `duration` intended to be an optional query parameter? If so, you’ll
need to provide a default here (`request.args.get('duration', '0')`), or
the `int(None)` conversion will be a type error. If it’s intended to be
mandatory, please use `request.args['duration']` to make that clear.

Reply:
Done, set a default duration to 1s.